integration-test/src/test/java/org/apache/iotdb/confignode/it/cluster/IoTDBClusterNodeErrorStartUpIT.java

```diff
@@ -297,4 +301,47 @@ public void testIllegalNodeRestart()
               Arrays.asList(NodeStatus.Running, NodeStatus.Running));
     }
   }
+
+  @Test
+  public void testIllegalNodeStartUp()
+      throws StartupException, ConfigurationException, IOException, ClientManagerException,
+          InterruptedException, TException {
+    ConfigNodeWrapper portConflictConfigNodeWrapper =
+        EnvFactory.getEnv().generateRandomConfigNodeWrapper();
+    DataNodeWrapper portConflictDataNodeWrapper =
+        EnvFactory.getEnv().generateRandomDataNodeWrapper();
+    try (SyncConfigNodeIServiceClient client =
+        (SyncConfigNodeIServiceClient) EnvFactory.getEnv().getLeaderConfigNodeConnection()) {
+      // set ConfigNode port repeat
+      TShowClusterResp showClusterResp = client.showCluster();
+      int beforeStartConfigNodes = showClusterResp.getConfigNodeListSize();
+      portConflictConfigNodeWrapper.setConsensusPort(portConflictConfigNodeWrapper.getPort());
+      portConflictConfigNodeWrapper.changeConfig(
+          (MppBaseConfig) EnvFactory.getEnv().getConfig().getConfigNodeConfig(),
+          (MppCommonConfig) EnvFactory.getEnv().getConfig().getConfigNodeCommonConfig(),
+          null);
+      portConflictConfigNodeWrapper.start();
+      for (int i = 0; i < START_RETRY_NUM; ++i) {
+        showClusterResp = client.showCluster();
+        Thread.sleep(1000);
+      }
+      int afterStartConfigNodes = showClusterResp.getConfigNodeListSize();
+      Assert.assertEquals(beforeStartConfigNodes, afterStartConfigNodes);
+      // set datanode port repeat
+      int beforeStartDataNodes = showClusterResp.getDataNodeListSize();
+      portConflictDataNodeWrapper.setMppDataExchangePort(
+          portConflictDataNodeWrapper.getDataRegionConsensusPort());
+      portConflictDataNodeWrapper.changeConfig(
+          (MppBaseConfig) EnvFactory.getEnv().getConfig().getDataNodeConfig(),
+          (MppCommonConfig) EnvFactory.getEnv().getConfig().getDataNodeCommonConfig(),
+          null);
+      portConflictDataNodeWrapper.start();
+      for (int i = 0; i < START_RETRY_NUM; ++i) {
+        showClusterResp = client.showCluster();
+        Thread.sleep(1000);
+      }
+      int afterStartDataNodes = showClusterResp.getDataNodeListSize();
+      Assert.assertEquals(beforeStartDataNodes, afterStartDataNodes);
```

Comment:
Either assertEquals of DataNodes within each loop or just sleep 10s here.

Reply:
fixed